Significant-Gravitas/AutoGPT
# [Fix Bug] Voice Ava (Premium) not found, add dynamic module for macos tts

### Background
This pull request aims to improve the macOS TTS voice support in the Auto-GPT repository. The previous implementation had a fixed voice for the second agent, "Ava (Premium)", which might not be available on all systems. This PR provides a more dynamic approach to handling macOS TTS voices.

### Changes
1. The MacOSTTS class has been updated to include a `_setup()` method that fetches the available voices on the system and sets up the default and custom voices accordingly.
2. Added a `get_voice_names()` method to the MacOSTTS class to retrieve available voice names on the system.
3. Updated the `_speech()` method in the MacOSTTS class to use voices from the `_voices` list instead of hardcoding voice names.
4. Updated the `.env.template` and `config.py` files to include new environment variables for specifying custom macOS TTS voices.

### Documentation
In-code comments and descriptions have been added to the MacOSTTS class methods, explaining their purpose, functionality, input arguments, and return values.

### Test Plan
1. Set up the Auto-GPT repository on a macOS system.
2. Configure the `.env` file to enable macOS TTS support and specify custom voices using the provided environment variables.
3. Run the Auto-GPT application and test with various prompts to ensure that the macOS TTS voices are working correctly and switching between different voices as expected.

### PR Quality Checklist
